### PR TITLE
feature: create api-languages

### PR DESCRIPTION
**Pull request by tataihono**
# Description

This PR adds in a new api-languages project. Apart from the graphql resolvers to display this data externally, two other key things are added in this PR.

- import languages from arclight (ARCLIGHT_API_KEY needed to carry out this operation)
- add a translation field decorator to simplify the filtering of translation fields and array of translation fields
- [Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26244172/todos/4694463991)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Add ARCLIGHT_API_KEY (@tataihono can give you this) and run nx seed api-languages. This should populate arango with a few hundred languages.
- [ ] Call the GraphQL gateway to return a list of languages
- [ ] Filter name by languageId 529 and primary true

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "import/newline-after-import": "error",
     "import/no-named-default": "error",
     "import/no-anonymous-default-export": "error",
-    "import/dynamic-import-chunkname": "error"
+    "import/dynamic-import-chunkname": "error",
+    "@typescript-eslint/no-empty-function": { "allow": ["decoratedFunctions"] }
```
Comment (tataihono) [inline .eslintrc.json:30]: this is needed to allow complex decorators that can be used across apis

File: apps/api-languages/db/seed.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  } catch {}
+  const mediaLanguages = await getMediaLanguages()
+
+  for (const mediaLanguage of mediaLanguages) {
+    console.log('language:', mediaLanguage.languageId)
+    await digestMediaLanguage(mediaLanguage)
+  }
+
+  for (const mediaLanguage of mediaLanguages) {
+    await digestMediaLanguageMetadata(mediaLanguage)
```
Comment (tataihono) [inline apps/api-languages/db/seed.ts:125]: metadata language tag is not in id format. Needs to be converted from bcp47 code to id.

File: libs/nest/decorators/src/lib/TranslationField/TranslationField.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+export interface Translation {
+  languageId: string
+  primary: boolean
+  value: string
+}
+
+export function TranslationField(name: string) {
```
Comment (tataihono) [inline libs/nest/decorators/src/lib/TranslationField/TranslationField.ts:7]: this function handles an array of translations and an array of array of translations

File: apps/api-languages/.env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+DATABASE_URL="arangodb://arangodb:8529"
```
Comment (mikeallisonJS) [inline apps/api-languages/.env:1]: Shouldn't include .env files

File: apps/api-languages/db/seed.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+async function main(): Promise<void> {
+  try {
+    await db.createCollection('languages', { keyOptions: { type: 'uuid' } })
+  } catch {}
+  const mediaLanguages = await getMediaLanguages()
+
+  for (const mediaLanguage of mediaLanguages) {
+    console.log('language:', mediaLanguage.languageId)
```
Comment (mikeallisonJS) [inline apps/api-languages/db/seed.ts:120]: leftover console.log?

File: apps/api-languages/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,8 @@
+FROM node:14-alpine
+WORKDIR /app
+COPY ./dist/apps/api-languages .
+EXPOSE 4002
```
Comment (mikeallisonJS) [inline apps/api-languages/Dockerfile:4]: 4003?

**Review by mikeallisonJS — DISMISSED** [review]

File: apps/api-languages/db/seed.ts (hunk 5 above)
Comment (tataihono) [reply apps/api-languages/db/seed.ts:120]: put the console log into the arclight seed import so the developer could see that something was happening.

Comment (tataihono) [reply apps/api-languages/db/seed.ts:120]: put the console log into the arclight seed import so the dev could see that something was happening.

File: apps/api-languages/.env (hunk 4 above)
Comment (tataihono) [reply apps/api-languages/.env]: fixed

File: apps/api-languages/Dockerfile (hunk 6 above)
Comment (tataihono) [reply apps/api-languages/Dockerfile]: fixed.

**Review by mikeallisonJS — APPROVED** [review]

**Pull request merged**